tests/src/Context/TaxonomyContext.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\nexteuropa\Context\TaxonomyContext.
+ */
+
+namespace Drupal\nexteuropa\Context;
+
+use Behat\Behat\Context\Context;
+
+/**
+ * Context with Taxonomy functionality.
+ */
+class TaxonomyContext implements Context {
+
+  /**
+   * List of Vocabularies created during test execution.
+   *
+   * @var \Vocabulary[]
+   */
+  protected $vocabularies = [];
+
+  /**
+   * List of Fields created during test execution.
+   *
+   * @var \Fields[]
+   */
+  protected $fields = [];
+
+  /**
+   * Create vocabulary.
+   *
+   * @param string $name
+   *    Name of the taxonomy.
+   *
+   * @When I create a new vocabulary :name
+   */
+  public function iCreateNewVocabulary($name) {
+    $vocabulary = array(
+      'name' => $name,
+      'machine_name' => $name,
+      'description' => '',
+      'module' => 'taxonomy',
+    );
+    taxonomy_vocabulary_save((object) $vocabulary);
+
+    // Save the vocabulary for cleaning.
+    $this->vocabularies[] = $name;
+  }
+
+  /**
+   * Revert to previous settings after scenario execution.
+   *
+   * @AfterScenario
+   */
+  public function removeVocabularies() {
+    // Remove the vocabularies.
+    foreach ($this->vocabularies as $vocabulary_name) {
+      $vid = db_query("SELECT vid FROM {taxonomy_vocabulary} WHERE machine_name = '$vocabulary_name'")->fetchField();
+      taxonomy_vocabulary_delete($vid);
+    }
+  }
+
+  /**
+   * Create term in a vocabulary.
+   *
+   * @param string $term_name
+   *    Name of the term.
+   * @param string $vocabulary_name
+   *    Name of the vocabulary.
+   *
+   * @Then I create a new term :term_name in the vocabulary :vocabulary_name
+   */
+  public function iCreateNewTermInTheVocabulary($term_name, $vocabulary_name) {
+    // Get the vocabulary ID.
+    $vid = db_query("SELECT vid FROM {taxonomy_vocabulary} WHERE machine_name = '$vocabulary_name'")->fetchField();
```

Comment:
Second time where this query is done, a helper function combined with Taxonomy functions can simplify the code.